experimental/casemapping/src/casemapping.rs

```diff
@@ -0,0 +1,149 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[cfg(feature = "provider_transform_internals")]
+use crate::error::Error;
+use crate::internals::*;
+#[cfg(feature = "provider_transform_internals")]
+use crate::provider::CaseMappingV1;
+use crate::provider::CaseMappingV1Marker;
+#[cfg(feature = "provider_transform_internals")]
+use icu_codepointtrie::CodePointTrieHeader;
+use icu_locid::Locale;
+use icu_provider::prelude::*;
+
+/// A struct with the ability to convert characters and strings to uppercase or lowercase,
+/// or fold them to a normalized form for case-insensitive comparison.
+///
+/// TODO: Expand documentation
+#[derive(Clone)]
+pub struct CaseMapping {
+    internals: DataPayload<CaseMappingV1Marker>,
+    locale: CaseMapLocale,
+}
+
+impl CaseMapping {
+    /// A constructor which takes a [`ResourceProvider`] and creates a [`CaseMapping`].
+    pub fn new<P>(provider: &P) -> Result<CaseMapping, DataError>
```

Comment:
Suggestion (optional): Consider changing the name to `try_new_default()` or `try_new_global()` or something similar to indicate the assumption that this method is making.  If you can't come up with a sufficiently good name then leave it as `try_new()`.

Reply:
I was thinking of this along the same lines as `String`, although there it's `String::new` and String::with_capacity` (not `new_with_capacity`). I have a slight preference for leaving this as `try_new` and describing the default locale behaviour in the doc comment instead of trying to fit it in the name, but if there are other cases in ICU4X where we use something like "try_new_default" then consistency is probably enough of an argument to sway me.